Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/LoginDialog.kt

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.features.settings.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.launch
+import org.dolphinemu.dolphinemu.databinding.DialogLoginBinding
+import org.dolphinemu.dolphinemu.dialogs.AlertMessage
+import org.dolphinemu.dolphinemu.features.settings.model.AchievementModel.asyncLogin
+import org.dolphinemu.dolphinemu.features.settings.model.NativeConfig
+import org.dolphinemu.dolphinemu.features.settings.model.StringSetting
+
+class LoginDialog : DialogFragment() {
+    private var _binding: DialogLoginBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = DialogLoginBinding.inflate(inflater, container, false)
+        binding.usernameInput.setText(
+            StringSetting.ACHIEVEMENTS_USERNAME.string
+        )
+        return binding.getRoot()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.buttonCancel.setOnClickListener { onCancelClicked() }
+        binding.buttonLogin.setOnClickListener { onLoginClicked() }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun onCancelClicked() {
+        dismiss()
+    }
+
+    private fun onLoginClicked() {
+        StringSetting.ACHIEVEMENTS_USERNAME.setString(NativeConfig.LAYER_BASE_OR_CURRENT,
+            binding.usernameInput.text.toString())
+        lifecycleScope.launch {
+            if (asyncLogin(binding.passwordInput.text.toString()))
+                dismiss()
+            else
+                AlertMessage.newInstance("Login Failed", "Login Failed", false, false)
```

Comment:
These strings aren't translatable.

I don't recommend using AlertMessage here - it's intended only as an implementation detail of the panic alert system. `MaterialAlertDialogBuilder` is a better fit, and it also automatically handles translation if you use a string resource.

Reply:
Or if you want to do something that's a bit nicer for the user - letting them skip having to dismiss the dialog - you could have a "login failed" UI element in LoginDialog's layout that's hidden by default and becomes shown if login fails.